sources/common/inc/Logger.h

```diff
@@ -46,32 +46,42 @@ namespace LOGGER
     class Logger
     {
         public:
-            /// The logger constructor.
-            Logger()
-            {}
-
-            /// The destructor.
-            ~Logger();
-
             /// The warning logging method.
             void LogWarning(uint32_t number, const std::string &rMessage);
 
             /// The error logging method.
-            void Log(const EXCEPTION::VcdException &rException);
+            void LogError(const EXCEPTION::VcdException &rException);
+
+            /// Let the logger be a singleton.
+            static Logger &GetInstance()
+            {
+                static Logger instance;
+                return instance;
+            }
+
+            /// Sets the output stream.
+            /// The method is supposed for unit testing.
+            ///
+            /// @param pOutputStream The default outputstream for warnings and errors.
+            void SetOutput(std::ostream *pOutputStream);
```

Comment:
That's because the class is a singleton and the constructor cannot be called explicitly. So by default std::cerr is used but it can be redirected. Because the output stream cannot be initialized by a constructor and its value changes the pointer has been used.

We could do it via public constructor but it wouldn't be a singleton. It seemed to me a cleaner solution. But of course I'm open to new, fresh ideas :)